### PR TITLE
fix config file in `$HOME` dir and other refactors

### LINKED ISSUE #26: Global meteor.json
### 🐛 Describe the bug

The config file  `.meteor.json` is not found when placed in `$HOME`. Everything works as expected, when `.meteor.json` is placed above the working dir (even deeply nested) until  the uppermost dir is the home directory. 

It would be nice to have a global configuration in `~/.meteor.json` and/or `~/.config/meteor.json` and/or `~/.config/meteor/meteor.json`

### Versions

meteor version 0.18.0

### PR DESCRIPTION
**Pull request by sebastianwebber**
In summary, I moved some functions to a config package to extract the main package's config responsibility.

I also added some logger config to help troubleshoot the config file locations, which will work like this:
image

It will use the first file found like this:
image

Feel free to suggest any necessary changes.

It may close #26.

### CODE REVIEW

**Comment by sebastianwebber** [comment]
also fixed the issue that passed the flag args to the git command:

```shell
❯ go run main.go git.go config.go -L debug -- -s -S
2024-03-20 03:07:57 DEBU checking for config file path=/Users/seba/projetos/github.com/sebastianwebber/meteor/.meteor.json
2024-03-20 03:07:57 DEBU checking for config file path=/Users/seba/projetos/github.com/sebastianwebber/.meteor.json
2024-03-20 03:07:57 DEBU found config file path=/Users/seba/projetos/github.com/sebastianwebber/.meteor.json
2024-03-20 03:07:57 DEBU loading config file path=/Users/seba/projetos/github.com/sebastianwebber/.meteor.json
[fix-config-file d6dc116] fix: ignore non-parsed args
 1 file changed, 1 insertion(+), 1 deletion(-)

```

**Comment by sebastianwebber** [comment]
ideally we could use cobra to handle the command line args, let me know if you would like a PR to handle this.

**Comment by stefanlogue** [comment]
This looks good, I'll just need to get some time (hopefully this evening) to pull it down and give it a quick test. It may close #26, but this isn't the fix I had in mind for that issue. Obviously, if this does fix it then even better!

In regards to using Cobra to handle the command line arguments, I'd like to avoid adding unnecessary complexity to the tool as much as possible for now. If you can give me a good enough reason to make the move at this stage, then I'm all ears! But I feel like it's something I'd rather do when the complexity it adds is more justified

File: main.go [hunk 1]
```diff
@@ -34,16 +36,37 @@ func isFlagPassed(name string) bool {
...
-var version = "dev"
+var (
+	version  = "dev"
+	logLevel string
+)
 
-func main() {
+func init() {
 	flag.BoolP("version", "v", false, "show version")
+	flag.StringVarP(&logLevel, "log-level", "L", "info", "Log level (debug, info, warn, error, fatal, panic)")
```
Comment (stefanlogue) [inline main.go:46]: Do we need all of these different debug levels? I think having a flag for debugging is a great idea (I wish I had this when I was originally writing the code for loading the config files!!), I think we maybe only need to be able to toggle debugging on or off, which could be achieved like `meteor --debug` or `meteor -D`, as `git commit` does not use the `--debug` or `-D` flags

File: main.go [hunk 2]
```diff
@@ -231,7 +254,7 @@ func main() {
 		newCommit.Body = newCommit.Body + buildCoauthorString(newCommit.Coauthors)
 	}
 
-	args := os.Args[1:]
+	args := flag.Args()
```
Comment (stefanlogue) [inline main.go:257]: I wasn't aware of this function in the `pflags` package, nice!

File: main.go (hunk 1 above)
Comment (sebastianwebber) [reply main.go:46]: let me update that then. hold on.

Comment (sebastianwebber) [reply main.go:46]: Added some changes:
image

Comment (sebastianwebber) [reply main.go:46]:
```
❯ go run main.go git.go config.go --help
Usage of /var/folders/yx/7rrwmnvd3d14_qc362y5yt300000gn/T/go-build713247778/b001/exe/main:
  -D, --debug     enable debug mode
  -v, --version   show version
pflag: help requested
exit status 2
```

Comment (sebastianwebber) [reply main.go:46]: About cobra, let's discuss this in a new PR or issue.

Comment (stefanlogue) [reply main.go]: > Added some changes: image

This looks good, but I'm wondering if we want to raise a warning when a user has no config file? I imagine the vast majority of users won't need a config file at all if they just want to do vanilla conventional commits

Comment (sebastianwebber) [reply main.go]: ok, let's move to debug then.

Comment (sebastianwebber) [reply main.go]: 807b62578ad1ed027865ad910586b1651155c759 should fix the problem.

**Review by stefanlogue — APPROVED** [review]
LGTM!

**Pull request merged**